go.mod

```diff
@@ -44,18 +44,18 @@ require (
 	golang.org/x/mod v0.14.0
 	golang.org/x/text v0.14.0
 	gopkg.in/yaml.v3 v3.0.1
-	k8s.io/api v0.28.3
-	k8s.io/apimachinery v0.28.3
-	k8s.io/apiserver v0.28.3
-	k8s.io/client-go v0.28.3
-	k8s.io/cluster-bootstrap v0.27.2
-	k8s.io/component-base v0.28.3
+	k8s.io/api v0.28.4
```

Comment:
We could probably stand to bump `KUBEBUILDER_ENVTEST_KUBERNETES_VERSION` in the Makefile to 1.28 alongside these other k8s version changes.

Reply:
https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/4182/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R36